src/AutoMerge/guidelines.jl

```diff
@@ -322,6 +322,38 @@ function meets_name_match_check(
     return (true, "")
 end
 
+# This check checks for an explanation of why a breaking change is breaking
+const guideline_breaking_explanation = Guideline(;
+    info = "Release notes have not been provided that explain why this is a breaking change.",
+    docs = "If this is a breaking change, release notes must be given that explain why this is a breaking change (i.e. mention \"breaking\"). To update the release notes, please see the \"Providing and updating release notes\" subsection under \"Additional information\" below.",
+    check=data -> meets_breaking_explanation_check(data))
+
+function meets_breaking_explanation_check(data::GitHubAutoMergeData)
+    # Look up PR here in case the labels are slow to be applied by the Registrator bot
+    # which decides whether to add the BREAKING label
+    pr = GitHub.pull_request(data.api, data.registry, data.pr.number; auth=data.auth)
+    return meets_breaking_explanation_check(pr.labels, pr.body)
+end
+function meets_breaking_explanation_check(labels::Vector, body::AbstractString)
+    if any(==("BREAKING"), labels)
+        release_notes = get_release_notes(body)
+        if release_notes === nothing
+            return false, "This is a breaking change, but no release notes have been provided."
+        elseif !occursin(r"breaking", lowercase(release_notes))
+            return false, "This is a breaking change, but the release notes do not mention it."
```

Comment:
BTW this might be one that could benefit from an extended explanation of how to do it, like https://github.com/JuliaRegistries/RegistryCI.jl/blob/f74fef6646903b00992b10a7b9e98f123c4762f9/src/AutoMerge/guidelines.jl#L93-L117